src/main/kotlin/it/pagopa/ecommerce/payment/requests/services/PaymentRequestsService.kt

```diff
@@ -128,6 +133,13 @@ class PaymentRequestsService(
       return@flatMap paymentRequestInfo
     }
 
+  private fun traceNodoError(faultCode: String?) {
+    faultCode?.let { code ->
+      openTelemetryUtils.addSpanWithAttributes(
+        "nodoError", Attributes.of(AttributeKey.stringKey("faultCode"), code))
+    }
+  }
```

Comment:
I would let the caller handle the `null` case

```suggestion
  private fun traceNodoError(faultCode: String) {
    openTelemetryUtils.addSpanWithAttributes(
      "nodoError", Attributes.of(AttributeKey.stringKey("faultCode"), faultCode))
  }
```